test/future_tests.cpp

```diff
@@ -671,8 +671,7 @@ BOOST_AUTO_TEST_CASE(future_reduction_with_mutable_void_task) {
     BOOST_TEST_MESSAGE("future reduction with mutable task");
 
     std::atomic_int check{0};
-    auto func = [i = int{0}, &check]() mutable {
```

Comment:
This might be intentionally unused - but the compiler complained. I'm not sure what the purpose of it would be, though. @FelixPetriconi @sean-parent if you wrote this test and want this variable around, let me know.